### PR TITLE
Additional functions for build user friendly role management interface

### PR DESCRIPTION
**Pull request by dkhlystov**

### CODE REVIEW

**Comment by samdark** [comment]
@dkhlystov could you add these methods to interface and implement PHP manager as well as tests?

**Comment by dkhlystov** [comment]
When testing Php Manager a bug was found with determine users by role. Php automaticaly converts string array key to integer when I'm assigning role to user. So === checking is false and getUserIdsByRole returns wrong result.

**Comment by dkhlystov** [comment]
Is this realy bug?

**Comment by dkhlystov** [comment]
```
$role = $auth->getRole('page');
$auth->assign($role, 1);
var_dump($auth->getUserIdsByRole('page')); //array(1) { [0]=> string(1) "1" }
$auth->revoke($role, 1);
$auth->assign($role, '1');
var_dump($auth->getUserIdsByRole('page')); //array(0) { }
$auth->revoke($role, 1);

```

**Comment by dkhlystov** [comment]
It sems that is bug after some research how php working with arrays.

File: framework/rbac/ManagerInterface.php [hunk 1]
```diff
@@ -129,6 +129,14 @@ public function getRule($name);
     public function getRules();
 
     /**
+     * Checks the possibility of adding a child to parent
+     * @param Item $parent the parent item
+     * @param Item $child the child item to be added to the hierarchy
+     * @return boolean possibility of adding
+     */
+    public function canAddChild($parent, $child);
```
Comment (cebe) [inline framework/rbac/ManagerInterface.php]: @samdark Isn't adding methods to interfaces a BC break?

Comment (samdark) [inline framework/rbac/ManagerInterface.php]: It is but in this case I'd make an exception.

**Comment by samdark** [comment]
Thanks!

**Pull request closed without merging**